### DISCUSSION TITLE
oven/bun hangs in local dockerfile

### DISCUSSION THREAD

**Discussion by igortas** [oven-sh/bun#9114] [Q&A]
Hi,
I'm still having issues making proper testing ```bun``` with ```dockerfile```.

The process hangs on the ```build``` command and never ends.

![image](https://github.com/oven-sh/bun/assets/16512083/f76718bb-9520-4965-91af-4b461a92bdf5)

This is the last debug messages where bun hangs. 

This is not happening if I run ```bun run build``` without ```Dockerfile```.

![image](https://github.com/oven-sh/bun/assets/16512083/90f1204f-2b76-4c6e-916d-26c45cc2cba0)

```Dockerfile``` simple commands:

```
FROM oven/bun AS build
WORKDIR /app

COPY package.json bun.lockb ./

RUN bun install --frozen-lockfile

COPY . .

RUN bun run build
```

Happens on ```WSL2``` on Windows ```11```.

- Docker version: 25.0.3 installed on ```WSL2```
- Bun version locally 1.0.29

BR,

**Comment by marcosgomesneto** [comment]
Why don't you use DevContainers for Bun? Everything is already configured for Docker. Follow the link:

https://github.com/marcosgomesneto/bun-devcontainers

**Comment by igortas** [comment]
@marcosgomesneto I'm trying to replace ```bun``` with ```npm``` in very simple ```dockerfile```, so devcontainers is not the solution I want. i need to work in ci/cd

**Comment by matoruru** [comment]
@igortas What does "hang" mean exactly? Did you get any error messages on hanged? 🤔

**Comment by igortas** [comment]
@matoruru not really just hangs. I've try to comment the ```build``` command, so ```dockerfile``` can finish and than ```sh``` into the container, but still ```vite build``` gets stucked from inside the container.

**Comment by matoruru** [comment]
@igortas It sounds like Vite related issue not actually Bun. This situation seems to be happening with just Node.js and npm as well.

I think you can refer this post:
https://stackoverflow.com/questions/75839993/vite-build-hangs-forever
